From d928888cee80700ddf6ff57de189c12440368fcd Mon Sep 17 00:00:00 2001
From: Sebastian Krzyszkowiak <dos@dosowisko.net>
Date: Thu, 16 May 2019 19:09:01 +0200
Subject: [PATCH 8/9] cmake: make condition that excludes test_big_endian test
 more generic

When cross-compiling to some platforms, the test doesn't work. Testing
for variables set earlier instead of hardcoded set of platforms allows
one to skip the test by appending proper variable to cmake command line
or platform toolchain file.
---
 CMakeLists.txt | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f81c8979..7fd004974 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -313,6 +313,7 @@ if(IPHONE)
     set(ALLEGRO_CFG_PTHREADS_TLS 1)
     set(ALLEGRO_IPHONE 1)
     set(ALLEGRO_UNIX 0)
+    set(ALLEGRO_LITTLE_ENDIAN 1)
     set(WANT_X11 off)
     set(GL_AUTO_BUILD_TYPE "gles2+")
     set(WANT_GLES3 yes)
@@ -339,6 +340,7 @@ endif(ALLEGRO_RASPBERRYPI)
 
 if(EMSCRIPTEN)
    set(GL_AUTO_BUILD_TYPE "gles2+")
+   set(ALLEGRO_LITTLE_ENDIAN 1)
    if (NOT ALLEGRO_SDL)
       message(FATAL_ERROR
          "Emscripten support currently requires the SDL backend (set ALLEGRO_SDL)")
@@ -462,13 +464,12 @@ include(FindPkgConfig)
 
 include(TestBigEndian)
 
-if(NOT IPHONE AND NOT EMSCRIPTEN)
+if(NOT ALLEGRO_BIG_ENDIAN AND NOT ALLEGRO_LITTLE_ENDIAN)
     test_big_endian(ALLEGRO_BIG_ENDIAN)
-endif(NOT IPHONE AND NOT EMSCRIPTEN)
-
-if(NOT ALLEGRO_BIG_ENDIAN)
-    set(ALLEGRO_LITTLE_ENDIAN 1)
-endif(NOT ALLEGRO_BIG_ENDIAN)
+    if(NOT ALLEGRO_BIG_ENDIAN)
+        set(ALLEGRO_LITTLE_ENDIAN 1)
+    endif(NOT ALLEGRO_BIG_ENDIAN)
+endif(NOT ALLEGRO_BIG_ENDIAN AND NOT ALLEGRO_LITTLE_ENDIAN)
 
 check_include_files(dirent.h ALLEGRO_HAVE_DIRENT_H)
 check_include_files(inttypes.h ALLEGRO_HAVE_INTTYPES_H)
-- 
2.22.0

